src/openrct2/core/FileScanner.cpp

```diff
@@ -83,32 +83,29 @@ class FileScannerBase : public IFileScanner
         _recurse = recurse;
         _patterns = GetPatterns(Path::GetFileName(pattern));
 
-        _currentPath = Memory::Allocate<utf8>(MAX_PATH);
-        _currentFileInfo = Memory::Allocate<FileInfo>();
+        _currentPath.resize(MAX_PATH);
+        _currentFileInfo = std::make_unique<FileInfo>();
```

Comment:
```suggestion
        _currentFileInfo = FileInfo{};
```
No need to have unique_ptr